### PR TITLE
CSCFAIRMETA-511: Filepicker v2 for etsin

### PR DESCRIPTION
**Pull request by tahme**
* Reorganize code to allow Etsin use Qvain Light filepicker functionality

* Allow unauthenticated users to access v2 file read APIs for a public dataset

* Move metaxApiV2 flag to Env store

* Make Env a dependency of Qvain

* Aria-label fixes

* Fix changed metadata not being saved in some cases

Fixes for Save and Publish
* Fix Save and Publish failing for new datasets
* Show validation errors properly
* Fix error from using setState on unmounted SubmitButtons component

### CODE REVIEW

**Comment by tahme** [comment]
Like previously, enable the new picker with `localStorage.setItem('metax_api_v2',1)`. The old functionality should work as before.

File: etsin_finder/frontend/js/stores/view/common.files.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+  }
+
+  @action loadProjectInfo = async () => {
+    const identifier = this.datasetIdentifier
+    const alreadyLoading = this.loadingProjectInfo && this.loadingProjectInfo.identifier === this.identifier
+    if (alreadyLoading) {
+      return
+    }
+
+    if (this.loadingProjectInfo && this.loadingProjectInfo.promise && !this.loadingProjectInfo.error) {
```
Comment (ismotee) [inline etsin_finder/frontend/js/stores/view/common.files.js:68]: Like alkreadyLoading above these checks could be named eg. returnedError

File: etsin_finder/frontend/js/stores/view/datasetquery.js [hunk 2]
```diff
@@ -62,6 +68,11 @@ class DatasetQuery {
     })
   }
 
+  @action
+  getFilesV2() {
```
Comment (ismotee) [inline etsin_finder/frontend/js/stores/view/datasetquery.js:72]: Assuming this.Files.openDataset(this.results) will fetch something and store results into a variable, the name of the action "getFilesV2" is misleading. "get" suggests that this function is merely a getter but it really does much of other stuff. fetchAndStore would be better in this case or just fetch if storing is just a byproduct of the function (and arg is optional).

File: etsin_finder/frontend/js/components/dataset/data/idaResourcesV2/fileTreeItem.jsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  }
+
+  const haveMetadata = hasMetadata(item)
+  const infoColor = haveMetadata ? 'primary' : 'gray'
+  const { type } = item
+
+  return (
+    <ItemRow style={{ flexWrap: 'wrap' }}>
+      <Group>
+        <Translate
```
[the remark this answers is not in the document]
Comment (tahme) [reply etsin_finder/frontend/js/components/dataset/data/idaResourcesV2/fileTreeItem.jsx]: done

File: etsin_finder/frontend/js/components/dataset/data/idaResourcesV2/index.jsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+          onClick={() => downloadAll(props.dataset.identifier)}
+        >
+          <Translate content={'dataset.dl.downloadAll'} />
+          <Translate className="sr-only" content="dataset.dl.file_types.both" />
+          <DownloadIcon />
+        </HeaderButton>
+      </Header>
+
+      <Tree allowDownload={allowDownload} allowInfo={allowInfo} />
+      {inInfo && (
```
[the remark this answers is not in the document]
Comment (tahme) [reply etsin_finder/frontend/js/components/dataset/data/idaResourcesV2/index.jsx]: handled the props separately, done

File: etsin_finder/frontend/js/stores/view/common.files.js (hunk 1 above)
Comment (tahme) [reply etsin_finder/frontend/js/stores/view/common.files.js]: done

File: etsin_finder/frontend/js/stores/view/datasetquery.js (hunk 2 above)
Comment (tahme) [reply etsin_finder/frontend/js/stores/view/datasetquery.js]: done

File: etsin_finder/frontend/js/components/common/files/items.jsx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { tint } from 'polished'
 import { faChevronRight, faChevronDown } from '@fortawesome/free-solid-svg-icons'
```
[the remark this answers is not in the document]
Comment (tahme) [reply etsin_finder/frontend/js/components/common/files/items.jsx]: moved to general

**Pull request merged**